swarm/api/api.go

```diff
@@ -202,3 +212,151 @@ func (self *Api) Modify(uri, contentHash, contentType string, nameresolver bool)
 	}
 	return trie.hash.String(), nil
 }
+
+func (self *Api) AddFile(mhash, path, fname string, content []byte, nameresolver bool) (storage.Key, string, error) {
```

Comment:
You should consider pulling in my changes from #3773 where I have added a `ManifestWriter` object which makes it easier to add / remove files from a manifest, see [here](https://github.com/lmars/go-ethereum/blob/swarm-api-refactor/swarm/api/manifest.go#L69-L105).